cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,81 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
 
+import HomePageObject from '../support/pages/home.pageObject';
+import SignUpPageObject from '../support/pages/signup.pageObject';
+
+const signUpPage = new SignUpPageObject();
+const homePage = new HomePageObject();
+
 describe('Sign Up page', () => {
   before(() => {
+    cy.task('db:clear');
+  });
+
+  beforeEach(() => {
+    signUpPage.visit();
+  });
+
+  it('should register a new user successfully', () => {
+    const user = signUpPage.registerUser();
+    homePage.assertHeaderContainUsername(user.username);
+    signUpPage.verifyRegistrationSuccess(); // Перевірка успішності реєстрації
+  });
+
+  it('should show error when username is missing', () => {
+    signUpPage.registerUserWithMissingUsername(); // Виконуємо реєстрацію без юзернейма
+    signUpPage.verifyRegistrationError('Username field required');// Перевіряємо, що відобразилась помилка
+  });
+
+  it('should show error when email is missing', () => {
+    signUpPage.registerUserWithMissingEmail(); // Виконуємо реєстрацію без електронної пошти
+    signUpPage.verifyRegistrationError('Email field required'); // Перевіряємо, що відобразилась помилка
+  });
 
+  it('should show error when password is missing', () => {
+    signUpPage.registerUserWithMissingPassword(); // Виконуємо реєстрацію без пароля
```

Comment:
Ensure that the `registerUserWithMissingPassword` method in `SignUpPageObject` is implemented correctly and triggers the appropriate error message.